src/hooks/useDataState.ts

```diff
@@ -28,11 +37,13 @@ export default function useDataState(items: object[], multipler: number, {height
         set_dataSource(dataProvider.cloneWithRows([...fakeItems]))
         set_layoutProvider(new LayoutProvider(
             (index) => {
-                return 'FULL'
+                return layoutTypeResolver.current ? layoutTypeResolver.current(index % items.length): 'FULL'
```

Comment:
why do you use % here? the index is the direct position of the item in the data array.

Reply:
The data source which is being passed would of length x =  (multiplier * actualData.length + 1).
The layout provider getType would be called for (0 to x).